internal/cmdreporeg/command.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmdreporeg
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/GoogleContainerTools/kpt/internal/errors"
+	"github.com/GoogleContainerTools/kpt/internal/util/parse"
+	"github.com/GoogleContainerTools/kpt/internal/util/porch"
+	configapi "github.com/GoogleContainerTools/kpt/porch/controllers/pkg/apis/porch/v1alpha1"
+	"github.com/spf13/cobra"
+	coreapi "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	command = "cmdreporeg"
+	longMsg = `
+kpt alpha repo reg[ister] REPOSITORY [flags]
+
+Args:
+
+REPOSITORY:
+  Address of the repository to register. Required argument.
+
+Flags:
+
+--description
+  Brief description of the package repository.
+
+--name
+  Name of the package repository. If unspecified, will use the name portion (last segment) of the repository URL.
+
+--title
+  Title of the package repository.
+
+--deployment
+  Repository is a deployment repository; packages in a deployment repository are considered deployment-ready.
+
+--repo-username
+  Username for repository authentication.
+
+--repo-password
+  Password for repository authentication.
+`
+)
+
+func newRunner(ctx context.Context, rcg *genericclioptions.ConfigFlags) *runner {
+	r := &runner{
+		ctx: ctx,
+		cfg: rcg,
+	}
+	c := &cobra.Command{
+		Use:     "reg REPOSITORY",
```

Comment:
super nit: `reg [REPOSITORY]`

Reply:
Looking at the rest of `kpt` it seems that [...] signifies optional and this argument is required (for repository registration one must provide the repository address).
maybe I misunderstood, please LMK